apps/web/src/pages/quick-start/components/demo-flow/nodes/NodeStep.tsx

```diff
@@ -0,0 +1,175 @@
+import React, { useEffect, useState } from 'react';
+import styled from '@emotion/styled';
+import { Popover } from '@mantine/core';
+
+import { useStyles } from '../../../../../design-system/template-button/TemplateButton.styles';
+import { colors, shadows, Text } from '../../../../../design-system';
+import { guidePreview, GuideTitleEnum, IBeat, IGuide } from './consts';
+import { useSegment } from '../../../../../components/providers/SegmentProvider';
+import { When } from '../../../../../components/utils/When';
+
+export function NodeStep({
+  data,
+  id,
+  Handlers,
+  Icon,
+  ActionItem,
+}: {
+  data: any;
+  id: string;
+  Handlers: React.FC<any>;
+  Icon: React.FC<any>;
+  ActionItem?: React.FC<any>;
+}) {
+  const { theme } = useStyles();
+  const { counter } = useCounter();
+  const [sequence, setSequence] = useState<IBeat>();
+
+  const label = data.label.toLowerCase();
+  const popoverData = guidePreview[label] as IGuide;
+  const titleGradient =
+    popoverData.title === GuideTitleEnum.DIGEST_PREVIEW || popoverData.title === GuideTitleEnum.DIGEST_PLAYGROUND
+      ? 'blue'
+      : 'red';
+
+  useEffect(() => {
+    setSequence(popoverData.sequence[counter.toString()] as IBeat);
+  }, [counter]);
+
+  return (
+    <Popover
+      withArrow
+      opened={sequence?.open || false}
+      transition="rotate-left"
+      transitionDuration={600}
+      position="right"
+      radius="md"
+      shadow={theme.colorScheme === 'dark' ? shadows.dark : shadows.medium}
+      offset={30}
+    >
+      <Popover.Target>
+        <div>
+          <StepCard data-test-id={`data-test-id-${label}`}>
+            <ContentContainer>
+              <LeftContent>
+                <Icon style={{ marginRight: '15px' }} />
+                <Text weight={'bold'}>{data.label} </Text>
+              </LeftContent>
+              {ActionItem ? <ActionItem /> : null}
+            </ContentContainer>
+          </StepCard>
+          <Handlers />
+        </div>
+      </Popover.Target>
+      <Popover.Dropdown
+        style={{
+          height: '100px',
+          padding: '16px',
+          backgroundColor: theme.colorScheme === 'dark' ? colors.B17 : colors.white,
```

Comment:
Unfortunately i was not able to update the opacity of this element, actually will love to get some help with it 😃

Reply:
the opacity is controlled by the transition animation of the dropdown... in short, the styles and opacity are overridden by the lib: https://github.com/mantinedev/mantine/blob/master/src/mantine-core/src/Popover/PopoverDropdown/PopoverDropdown.tsx

need to think how we can workaround it...
so that's why you haven't implemented that opacity in animation right?

Reply:
That's right I was not able to update it...